src/Nuget.PackageIndex.VisualStudio/Analyzers/ProjectTargetFrameworkProviderExporter.cs

```diff
@@ -175,6 +176,16 @@ private void OnAfterSolutionClosing()
             ClearCache();
         }
 
+        private void OnProjectRemoved(Project project)
+        {
+            if (project == null)
```

Comment:
I added Debug.Assert for us to see it internally in debug builds, but still prefer to keep null check since if it happens on users machine we would never know, but users would see null exception dialog which is not good. (I checked in index , other subscribers also check for null in such event handlers)